src/app/shared/services/auth.service.ts

```diff
@@ -0,0 +1,46 @@
+import { Injectable } from '@angular/core';
+import {
+  Auth,
+  User,
+  authState,
+  createUserWithEmailAndPassword,
+  signInWithEmailAndPassword,
+  signOut,
+} from '@angular/fire/auth';
+import { BehaviorSubject, Observable, from, map, tap } from 'rxjs';
+import { IAuthCredentials } from '../models/auth-credentials.interface';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class AuthService {
+  private userSubject = new BehaviorSubject<User | null>(null);
+  user$ = this.userSubject.asObservable();
+
+  constructor(private auth: Auth) {}
```

Comment:
try to avoid constructor now and use inject

Reply:
http = inject(HttpClient);
like this